### PR TITLE
Fix all the failing tests

### PR DESCRIPTION
**Pull request by kamalcph**
- After this patch, no test failures were seen in the core module.

### CODE REVIEW

**Comment by kamalcph** [comment]
@satishd Please review this patch when you get time.

File: core/src/main/scala/kafka/log/Log.scala [hunk 1]
```diff
@@ -1721,41 +1721,35 @@ class Log(@volatile private var _dir: File,
...
+          } else {
+            None
+          }
+        }
 
-      if (isFirstSegment && remoteLogManager.isDefined) {
-        val localOffset = targetSeg.get.findOffsetByTimestamp(targetTimestamp, logStartOffset)
-        val remoteOffset = remoteLogManager.get.findOffsetByTimestamp(topicPartition, targetTimestamp, logStartOffset)
+        if (isFirstSegment && remoteLogManager.isDefined) {
+          val localOffset = targetSeg.get.findOffsetByTimestamp(targetTimestamp, localLogStartOffset)
```
Comment (kamalcph) [inline core/src/main/scala/kafka/log/Log.scala:1738]: Both log_start_offset and local_log_start_offset are valid as starting_offset. Added the latter for clarity.

Comment (satishd) [reply core/src/main/scala/kafka/log/Log.scala:1738]: This makes sense.

**Review by satishd — APPROVED** [review]
thanks @kamalcph for the PR, LGTM.

**Pull request merged**